runtime/caml/platform.h

```diff
@@ -414,14 +417,16 @@ Caml_inline unsigned caml_plat_spin_step(unsigned spins,
   }
 }
 
-#define SPIN_WAIT_BACK_OFF(max_spins)                                   \
-  unsigned CAML_GENSYM(spins) = 0;                                      \
-  unsigned CAML_GENSYM(max_spins) = (max_spins);                        \
-  static const struct caml_plat_srcloc CAML_GENSYM(loc) = {             \
-    __FILE__, __LINE__, __func__                                        \
-  };                                                                    \
-  for (; 1; CAML_GENSYM(spins) = caml_plat_spin_step(                   \
-         CAML_GENSYM(spins), CAML_GENSYM(max_spins), &CAML_GENSYM(loc)))
+#define SPIN_WAIT_BACK_OFF_(spins, max_spins, N, loc)           \
+  static const struct caml_plat_srcloc loc =                    \
+    { __FILE__, __LINE__, __func__ };                           \
+  for (unsigned int spins = 0, max_spins = (N);                 \
+       true;                                                    \
+       spins = caml_plat_spin_step(spins, max_spins, &loc))
```

Comment:
I think keeping the layout of the original is slightly clearer here:
```suggestion
  unsigned spins = 0;                                              \
  unsigned max_spins = (N);                                        \
  static const struct caml_plat_srcloc loc = {                     \
      __FILE__, __LINE__, __func__                                 \
      };                                                           \
  for (; true; spins = caml_plat_spin_step(spins, max_spins, &loc))
```
?

Reply:
My version has the advantage of limiting the scope of `spins` and `max_spins` to the loop.

Reply:
Hah - true, although I think the readability matters rather more, given that the scope of the variable is de facto limited to the macro. It's all subjective, though - I'm not pushing this one!

Reply:
I think I've done something better where the bound of the loop is now `const` and defined outside of it, but the iterator is defined inside the `for` clause.

Reply:
It's a beautiful compromise - you're definitely destined for a Standards Committee some day!

Reply:
I learned from the best ;)